src/MCPClient/lib/clientScripts/restructureDIPForContentDMUpload.py

```diff
@@ -22,614 +22,118 @@
-            dmdSecPair['nonDc'] = None
+    dmdSecPair = {'dc': None, 'nonDc': None}
+    if lenDmdSecs > 2:
+        print >> sys.stderr, 'Error splitting dmdSecs, more than 2 provided'
+        return dmdSecPair
+    for dmdSec in dmdSecs:
+        mdWrap = dmdSec.find('mets:mdWrap', namespaces=ns.NSMAP)
+        if mdWrap.get('MDTYPE') == 'OTHER':
+            dmdSecPair['nonDc'] = parseDmdSec(dmdSec)
+        if mdWrap.get('MDTYPE') == 'DC':
+            dmdSecPair['dc'] = parseDmdSec(dmdSec)
     if lenDmdSecs == 0:
         # If dmdSecs is empty, let parseDcXML() assign a placeholder title in dcMetadata.
-        dmdSec = dmdSecs
-        dmdSecPair['dc'] = parseDmdSec(dmdSec)
-        dmdSecPair['nonDc'] = None
+        dmdSecPair['dc'] = parseDmdSec(None)
 
     return dmdSecPair
 
 
-# Given a list of structMaps and a DMDID value, return a list of all the
-# <fptr> values for the files named in the structMap corresponding to
-# to the DMDID.
-def getFileIdsForDmdSec(structMaps, dmdSecIdValue):
-    dmdSecIdValue = dmdSecIdValue.strip()
-    fileIds = []
-    # We use the Archivematica default structMap, which is always the first.
-    structMap = structMaps[0]
-    for div in structMap.getElementsByTagName('div'):
-        for k, v in div.attributes.items():
-            # We match on the first dmdSec ID. Space is optional because 
-            # there could be two dmdSec IDs in the value, separated by a space.
-            match = re.search(r'%s\s?' % dmdSecIdValue, v)
-            if k == 'DMDID' and match:
-                for fptr in div.getElementsByTagName('fptr'):
-                    for k, v in fptr.attributes.items():
-                        if k == 'FILEID':
-                            fileIds.append(v)               
-    return fileIds
-
-
-# Given a group of dmdSecs and the METS structMaps, return a list of files
-# that are described by the dmdSecs.
-def getFilesInObjectDirectoryForThisDmdSecGroup(dmdSecGroup, structMaps):    
-    filesInObjectDirectoryForThisDmdSecGroup = list()
-    # Get the value of ID for each <dmdSec> and put them in a list,
-    # then pass the list into getFileIdsForDmdSec()
-    for dmdSec in dmdSecGroup:
-        Id = dmdSec.attributes['ID']
-        fileIds = getFileIdsForDmdSec(structMaps, Id.value)
-        for fileId in fileIds:
-            filename = getFptrObjectFilename(fileId, filesInObjectDirectory)
-            if filename is not None:
-                filesInObjectDirectoryForThisDmdSecGroup.append(filename)
-            
-    return filesInObjectDirectoryForThisDmdSecGroup
-
-
-# Add the AIP UUID to the DC metadata. We handle non-DC metadata within 
+# Add the AIP UUID to the DC metadata. We handle non-DC metadata within
 # each generateXXXProjectClientPackage function. Direct upload packages
 # have their DC metadata supplemented with the AIP UUID in generateDescFile().
 def addAipUuidToDcMetadata(dipUuid, dcMetadata):
+    if not dcMetadata:
```

Comment:
If there's no DC metadata provided (None), trying to add an identifier to it will fail. This returns the falsey value as is.  Would it be clearer if it returned None?